modules/openapi-generator/src/main/resources/php/HeaderSelector.mustache

```diff
@@ -42,31 +42,195 @@ class HeaderSelector
         if ($accept !== null) {
             $headers['Accept'] = $accept;
         }
-        if(!$isMultipart) {
+
+        if (!$isMultipart) {
             if($contentType === '') {
                 $contentType = 'application/json';
             }
+
             $headers['Content-Type'] = $contentType;
         }
 
         return $headers;
     }
 
     /**
-     * Return the header 'Accept' based on an array of Accept provided
+     * Return the header 'Accept' based on an array of Accept provided.
      *
      * @param string[] $accept Array of header
      *
      * @return null|string Accept (e.g. application/json)
      */
-    private function selectAcceptHeader($accept)
+    private function selectAcceptHeader(array $accept): ?string
     {
-        if (count($accept) === 0 || (count($accept) === 1 && $accept[0] === '')) {
+        # filter out empty entries
+        $accept = array_filter($accept);
+
+        if (count($accept) === 0) {
             return null;
-        } elseif ($jsonAccept = preg_grep('~(?i)^(application/json|[^;/ \t]+/[^;/ \t]+[+]json)[ \t]*(;.*)?$~', $accept)) {
-            return implode(',', $jsonAccept);
+        }
+
+        # If there's only one Accept header, just use it
+        if (count($accept) === 1) {
+            return reset($accept);
+        }
+
+		# If none of the available Accept headers is of type "json", then just use all them
```

Comment:
can you please replace tabs with 4-spaces for indention?

Reply:
Done! :smile:
Sorry for the mistake, 've been dealing with too many different coding standards recently... :stuck_out_tongue_winking_eye:

Reply:
No problem bro :)